### PR TITLE
Add support for Issuing APIs/resources

### PR DESCRIPTION
**Pull request by remi-stripe**
r? @ob-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: src/main/java/com/stripe/model/EventDataDeserializer.java [hunk 1]
```diff
@@ -44,9 +44,16 @@
     objectMap.put("invoice", Invoice.class);
     objectMap.put("invoice_line_item", InvoiceLineItem.class);
     objectMap.put("invoiceitem", InvoiceItem.class);
+    objectMap.put("issuing.authorization", com.stripe.model.issuing.Authorization.class);
+    objectMap.put("issuing.card", com.stripe.model.issuing.Card.class);
+    objectMap.put("issuing.cardholder", com.stripe.model.issuing.Cardholder.class);
+    objectMap.put("issuing.card_details", com.stripe.model.issuing.CardDetails.class);
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/EventDataDeserializer.java]: At the risk of sounding like a broken record, `CardDetails` after `Card` :)

File: src/main/java/com/stripe/model/issuing/Authorization.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+import com.stripe.exception.APIConnectionException;
+import com.stripe.exception.APIException;
+import com.stripe.exception.AuthenticationException;
+import com.stripe.exception.CardException;
+import com.stripe.exception.InvalidRequestException;
+import com.stripe.model.BalanceTransaction;
+import com.stripe.model.ExpandableField;
+import com.stripe.model.MetadataStore;
+import com.stripe.model.HasId;
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/issuing/Authorization.java]: Should be before `MetadataStore`

File: src/main/java/com/stripe/model/issuing/Card.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
+package com.stripe.model.issuing;
+
+import com.stripe.exception.APIConnectionException;
+import com.stripe.exception.APIException;
+import com.stripe.exception.AuthenticationException;
+import com.stripe.exception.CardException;
+import com.stripe.exception.InvalidRequestException;
+import com.stripe.model.Address;
+import com.stripe.model.MetadataStore;
+import com.stripe.model.HasId;
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/issuing/Card.java]: Same here.

File: src/main/java/com/stripe/model/issuing/Cardholder.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
+package com.stripe.model.issuing;
+
+import com.stripe.exception.APIConnectionException;
+import com.stripe.exception.APIException;
+import com.stripe.exception.AuthenticationException;
+import com.stripe.exception.CardException;
+import com.stripe.exception.InvalidRequestException;
+import com.stripe.model.Address;
+import com.stripe.model.MetadataStore;
+import com.stripe.model.HasId;
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/issuing/Cardholder.java]: And here.

File: src/main/java/com/stripe/model/issuing/Dispute.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
+package com.stripe.model.issuing;
+
+import com.stripe.exception.APIConnectionException;
+import com.stripe.exception.APIException;
+import com.stripe.exception.AuthenticationException;
+import com.stripe.exception.CardException;
+import com.stripe.exception.InvalidRequestException;
+import com.stripe.model.ExpandableField;
+import com.stripe.model.MetadataStore;
+import com.stripe.model.HasId;
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/issuing/Dispute.java]: And here.

File: src/main/java/com/stripe/model/issuing/Transaction.java [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+import com.stripe.exception.APIConnectionException;
+import com.stripe.exception.APIException;
+import com.stripe.exception.AuthenticationException;
+import com.stripe.exception.CardException;
+import com.stripe.exception.InvalidRequestException;
+import com.stripe.model.BalanceTransaction;
+import com.stripe.model.ExpandableField;
+import com.stripe.model.MetadataStore;
+import com.stripe.model.HasId;
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/issuing/Transaction.java]: And here.

File: src/main/java/com/stripe/model/issuing/MerchantData.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+@Getter
+@Setter
+@EqualsAndHashCode(callSuper = false)
+public final class MerchantData extends StripeObject {
+  String category;
+  String city;
+  String country;
+  String name;
+  String network_id;
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/issuing/MerchantData.java]: Should be `networkId`.

File: src/main/java/com/stripe/model/issuing/MerchantData.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  String postal_code;
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/issuing/MerchantData.java]: Should be `postalCode`.

File: src/main/java/com/stripe/net/APIResource.java [hunk 9]
```diff
@@ -77,11 +77,17 @@ public static void setStripeResponseGetter(StripeResponseGetter srg) {
...
     // Convert CamelCase to snake_case
-    final String className = clazz.getSimpleName()
+    String className = clazz.getSimpleName()
         .replaceAll("(.)([A-Z][a-z]+)", "$1_$2")
         .replaceAll("([a-z0-9])([A-Z])", "$1_$2")
         .toLowerCase();
 
+    // Issuing resources are in their own package. Until we can support adding OBJECT_NAME
+    // to all classes, we use this dirty trick to properly format the API endpoints
+    if(clazz.getName().contains("com.stripe.model.issuing.")) {
```
Comment (ob-stripe) [inline src/main/java/com/stripe/net/APIResource.java]: Need a space after `if`.

**Comment by remi-stripe** [comment]
@ob-stripe thank you for catching those! Fixed all of them, PTAL

**Comment by ob-stripe** [comment]
Released as 5.52.0.

**Pull request merged**